vertx-quickstart/src/main/java/org/acme/extra/FruitResource.java

```diff
@@ -32,40 +17,39 @@
 import javax.ws.rs.core.Response.ResponseBuilder;
 import javax.ws.rs.core.Response.Status;
 
+import io.quarkus.runtime.StartupEvent;
+import org.acme.extra.Fruit;
 import org.eclipse.microprofile.config.inject.ConfigProperty;
-import org.jboss.resteasy.annotations.jaxrs.PathParam;
 
 import io.smallrye.mutiny.Uni;
 import io.vertx.mutiny.pgclient.PgPool;
+import org.jboss.resteasy.reactive.RestPath;
 
 @Path("fruits")
 @Produces(MediaType.APPLICATION_JSON)
 @Consumes(MediaType.APPLICATION_JSON)
 public class FruitResource {
 
-    @Inject
-    @ConfigProperty(name = "myapp.schema.create", defaultValue = "true")
-    boolean schemaCreate;
+    private final PgPool client;
+    private final boolean schemaCreate;
 
-    @Inject
-    PgPool client;
+    public FruitResource(PgPool client, @ConfigProperty(name = "myapp.schema.create", defaultValue = "true") boolean schemaCreate) {
+        this.client = client;
+        this.schemaCreate = schemaCreate;
+    }
 
-    @PostConstruct
-    void config() {
+    private void initdb(@Observes StartupEvent ev) {
         if (schemaCreate) {
-            initdb();
+            client.query("DROP TABLE IF EXISTS fruits").execute()
+                    .flatMap(r -> client.query("CREATE TABLE fruits (id SERIAL PRIMARY KEY, name TEXT NOT NULL)")
+                            .execute())
+                    .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Orange')").execute())
+                    .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Pear')").execute())
+                    .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Apple')").execute())
+                    .await().indefinitely();
```

Comment:
@edeandrea this is similar to what you were asking me the other day

Reply:
Yeah I was looking for a way to do initialization like this in a non-blocking way.